### PR TITLE
[RE-234] Update AWS VM Creation

### PR DESCRIPTION
**Pull request by MikeyYeahYeah**
Updating the VM creation scripts to perform the following:
* Update the base image from Ubuntu 14.04 (Trusty) to Ubuntu 16.04 (Xenial)
* Bring these scripts inline with the machine-executor-images repository which is used to create the cloud images.

### CODE REVIEW

**Comment by MikeyYeahYeah** [comment]
Would it be possible to get a review from someone on @circleci/execution or @circleci/machine ?

File: circleci-provision-scripts/scala.sh [hunk 1]
```diff
@@ -7,7 +7,7 @@ function install_scala() {
     (cat <<'EOF'
 # Run sbt once to download dependencies.
 # SBT_OPTS="-XX:MaxMetaspaceSize=384M" sbt -v
-SBT_LAUNCH_VERSIONS="0.13.5 0.13.6 0.13.7 0.13.8 0.13.9 0.13.10 0.13.11 0.13.12 0.13.13 0.13.14 0.13.15"
+SBT_LAUNCH_VERSIONS="0.13.5 0.13.6 0.13.7 0.13.8 0.13.9 0.13.10 0.13.11 0.13.12 0.13.13"
```
Comment (glenjamin) [inline circleci-provision-scripts/scala.sh]: This looks like we're installing only older versions. I don't know much about SBT, but for a newer image it's probably the older ones which should be retired.

File: provision.sh [hunk 2]
```diff
@@ -1,46 +1,37 @@
...
+export CIRCLECI_USER=${CIRCLECI_USER:-circleci}
 
 set -ex
 
 cp circleci-install /usr/local/bin/circleci-install
 cp -r circleci-provision-scripts /opt/circleci-provision-scripts
 circleci-install base_requirements && circleci-install circleci_specific
 
-# Databases
-for package in mysql_57 mongo postgres; do circleci-install $package; done
```
Comment (glenjamin) [inline provision.sh]: These removals seem surprising - do you know why they're being removed?

File: provision.sh [hunk 3]
```diff
@@ -1,46 +1,37 @@
...
-
 # Installing java early beacuse a few things have the dependency to java (i.g. cassandra)
 circleci-install java oraclejdk8 && circleci-install java openjdk8
 
-for package in sysadmin devtools jq redis memcached rabbitmq ; do circleci-install $package; done
```
Comment (glenjamin) [inline provision.sh]: Same for these, i'd expect the services to be desirable

File: provision.sh [hunk 4]
```diff
@@ -1,46 +1,37 @@
...
-
-# Dislabe services by default
-for s in apache2 memcached rabbitmq-server neo4j neo4j-service elasticsearch beanstalkd cassandra riak couchdb mysql postgresql; do sysv-rc-conf $s off; done
```
Comment (glenjamin) [inline provision.sh]: Not starting services by default also seems desirable to me

File: provision.sh [hunk 5]
```diff
@@ -1,46 +1,37 @@
...
+circleci-install python 3.7.0
 sudo -H -i -u ${CIRCLECI_USER} pyenv global 2.7.12
 
 circleci-install nodejs 6.1.0
+circleci-install nodejs 8.11.2
+circleci-install nodejs 10.8.0
 sudo -H -i -u ${CIRCLECI_USER} nvm alias default 6.1.0
 
-circleci-install golang 1.7.3
+circleci-install golang 1.10.3
```
Comment (glenjamin) [inline provision.sh]: This is still very old

**Comment by MikeyYeahYeah** [comment]
@glenjamin Thank you for the great feedback.  Everything you are saying is on point.  However, I honestly don't have enough context around what is desirable state for a machine VM.  I simply mirrored the [machine-exector-images repo]( https://github.com/circleci/machine-executor-images)

Based off conversations I have had with machine team, it seems like ownership of that repository is in a state of flux.  I am not sure how to proceed quite honestly.

**Pull request merged**